src/util.ts

```diff
@@ -77,3 +79,18 @@ export async function exec(
     throw err;
   }
 }
+
+/**
+ * Log an error that occurred when performing some task on a workspace repo.
+ */
+export function logRepoError(err: Error, repo: WorkspaceRepo) {
+  const repoDirName = path.basename(repo.dir);
+  console.error(`${repoDirName}: ${err.message}`);
+}
+
+/**
+ * Log a user-facing message about progress through some set of steps.
+ */
+export function logStep(stepNum: string, emoji: string, msg: string) {
```

Comment:
every call is passing a string like `[1/2]`. I'd take the step numbers as arguments. It'll let you change formatting in one place, or start composing tasks by passing start and total step numbers around.